include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -1246,6 +1246,173 @@ namespace Differentiation
+     *     symbol "a" is substituted throughout the function first, and only
+     * then is the symbol "b(a)" substituted, by which time its explicit
+     * dependency on "a" cannot be resolved.
+     *  2. If <tt>map["a"] == "b"+2</tt> and <tt>map["b"] == 1</tt> then
+     *     then the function $f(a(b),a) = a+b$ will be evaluated and the result
+     *     $f\vert_{a=b+2, b} = [b+2+b]_{b=1} = 4$ is returned. This is because
+     * the explicitly dependent symbol "a(b)" is substituted first followed by
+     * the symbol "b".
+     */
+    Expression
+    substitute(const Expression &             expression,
+               const types::substitution_map &substitution_map);
+
+    /**
+     * Perform a substitution of the @p symbol into the given
+     * @p expression, with the result that all matches are assigned
+     * the corresponding @p value.
+     * This substitution function may be used to give a set of symbolic
+     * variables either a numeric interpretation or some symbolic definition.
+     *
+     * For more information regarding the performance of symbolic substitution,
+     * see the other @ref substitute(const Expression &,
+     * const types::substitution_map &) function.
+     *
+     * @note It is not required that all symbolic expressions be fully resolved
+     * when using this function. In other words, partial substitutions are
+     * valid.
+     *
+     * @tparam ValueType A type that corresponds to the @p value that the
+     *         @p symbol is to represent. Although it is typically
+     *         arithmetic in nature, it may also represent another symbolic
+     *         expression type.
+     */
+    template <typename ValueType>
+    Expression
+    substitute(const Expression &expression,
+               const Expression &symbol,
+               const ValueType & value);
+
+    /**
+     * Perform a single substitution sweep of a set of symbols into the given
+     * symbolic expression.
+     * The symbols in the @p expression that correspond to a matching entry key
+     * of the @p symbol_values vector entry are substituted by the entry's associated
+     * value.
+     * This substitution function may be used to give a set of symbolic
+     * variables either a numeric interpretation or some symbolic definition.
+     *
+     * For more information regarding the performance of symbolic substitution,
+     * see the other @ref substitute(const Expression &,
+     * const types::substitution_map &) function.
+     *
+     * @note It is not required that all symbolic expressions be fully resolved
+     * when using this function. In other words, partial substitutions are
+     * valid.
+     *
+     * @tparam ExpressionType A type that represents a symbolic expression.
+     *         The Differentiation::SD::Expression class is often suitable for
+     *         this purpose, but this may also represent Tensors and
+     *         SymmetricTensors of Expressions.
```

Comment:
I am not sure that `doxygen` creates proper references with the plural forms but don't have a better idea right now. Maybe, `Expressions->Expression objects`, `Tensors->Tensor objects` and `SymmetricTensors->SymmetricTensor objects`

Reply:
Yeah, unfortunately I've made this mistake throughout my documentation in this file and `symengine_tensor_operations.h`. Its going to take quite a bit of effort to correct. Perhaps I can defer than work until after the release? Not ideal I know, but I don't have the energy to do it tonight.

Reply:
Sure. That's not crucial for the release.